src/App.tsx

```diff
@@ -1,9 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setFilteredMovies] = useState(moviesFromServer);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+
+    const filteredMovies = moviesFromServer.filter(
+      movie => movie.title
+        .toLowerCase()
+        .includes(value.toLowerCase().trim())
+      || movie.description
+        .toLowerCase()
+        .includes(value.toLowerCase().trim()),
+    );
+
+    setFilteredMovies(filteredMovies);
```

Comment:
don't do it inside `handleChange` function, just set the correct query here
you may create `const filteredMovies` outside `handleChange` function and it will be recalculated (refiltered) every time `query` changes (on every state change)